### PR TITLE
Game stats sb

### PR DESCRIPTION
**Pull request by stellakunzang**
added methods for home_games, percentage_home_wins, percentage_visitor_wins, percentage_ties, count_of_games_by_season to stat_tracker class with reference to methods in classes

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -46,7 +41,24 @@ def create_game_teams
     @game_teams = GameTeam.from_csv(@game_teams_path)
   end
 
-  def all_teams
-    Team.all
+  def home_games
```
Comment (stellakunzang) [inline lib/stat_tracker.rb:44]: as discussed, added methods to stat_tracker class that call methods housed in other classes

File: test/fixtures/short_game_teams.csv [hunk 2]
```diff
@@ -2,3 +2,100 @@ game_id,team_id,HoA,result,settled_in,head_coach,goals,shots,tackles,pim,powerPl
 2012030221,3,away,LOSS,OT,John Tortorella,2,8,44,8,3,0,44.8,17,7
 2012030221,6,home,WIN,OT,Claude Julien,3,12,51,6,4,1,55.2,4,5
 2012030222,3,away,LOSS,REG,John Tortorella,2,9,33,11,5,0,51.7,1,4
+2012030222,6,home,WIN,REG,Claude Julien,3,8,36,19,1,0,48.3,16,6
```
Comment (stellakunzang) [inline test/fixtures/short_game_teams.csv:5]: agh ignore

File: test/fixtures/short_games.csv [hunk 3]
```diff
@@ -3,3 +3,99 @@ game_id,season,type,date_time,away_team_id,home_team_id,away_goals,home_goals,ve
 2012030222,20122013,Postseason,5/19/13,3,6,2,3,Toyota Stadium,/api/v1/venues/null
 2012030223,20122013,Postseason,5/21/13,6,3,2,1,BBVA Stadium,/api/v1/venues/null
 2012030224,20122013,Postseason,5/23/13,6,3,3,2,BBVA Stadium,/api/v1/venues/null
+2012030225,20122013,Postseason,5/25/13,3,6,1,3,Toyota Stadium,/api/v1/venues/null
```
Comment (stellakunzang) [inline test/fixtures/short_games.csv:6]: agh ignore

File: lib/game.rb [hunk 4]
```diff
@@ -12,6 +12,16 @@ def self.from_csv(csv_file_path)
...
+  def self.count_of_games_by_season
+    games_by_season = @@all.group_by { |game| game.season }
+    number_of_games_in_season =
+    count = {}
+    games_by_season.keys.each do |key|
+      count[key] = @@all.count { |game| game.season == key}
+    end
+    count
+  end
+
```
Comment (stellakunzang) [inline lib/game.rb:24]: I wrote a new method!

File: lib/game_team.rb [hunk 5]
```diff
@@ -11,6 +11,31 @@ def self.from_csv(csv_file_path)
...
+    ((visitor_wins / self.home_games) * 100).round(2)
+  end
+
+  def self.percentage_ties
+    games_count = @@all.count.to_f
+    ties_count = (@@all.find_all { |gt| gt.result == "TIE"}).count.to_f
+    ((ties_count / games_count) * 100).round(2)
+  end
+
+  def self.winningest_coach
```
Comment (stellakunzang) [inline lib/game_team.rb:34]: these are placeholders for methods I'm writing

File: lib/game.rb (hunk 4 above)
Comment (perryr16) [reply lib/game.rb:24]: so fun!

File: lib/game_team.rb [hunk 6]
```diff
@@ -11,6 +11,31 @@ def self.from_csv(csv_file_path)
     @@all = csv.map { |row| GameTeam.new(row) }
   end
 
+  def self.home_games
+    (@@all.find_all {|gt| gt.hoa == "home" }).count
+  end
+
```
Comment (perryr16) [inline lib/game_team.rb:17]: good abbrevaiation from game-team to gt

**Review by perryr16 — COMMENTED** [review]
n/a

**Review by Rostammahabadi — COMMENTED** [review]
This all looks good to me!

**Comment by foleymichelle9** [comment]
Agreed everything looks great!

**Pull request merged**